spec/controllers/hyrax/generic_works_controller_spec.rb

```diff
@@ -3,7 +3,73 @@
 require 'rails_helper'
 
 RSpec.describe Hyrax::GenericWorksController do
-  it "has tests" do
-    skip "Add your tests here"
+  routes { Rails.application.routes }
+  let(:main_app) { Rails.application.routes.url_helpers }
+  let(:hyrax) { Hyrax::Engine.routes.url_helpers }
+
+  describe '#manifest' do
+    let(:solr_document) { SolrDocument.new }
+    let(:ability) { nil }
+    let(:presenter) { Hyrax::GenericWorkPresenter.new(solr_document, ability, request) }
+    let(:mime2) { Hyrax::GenericWorksController::IIIF_DEFAULT_MANIFEST_MIME.gsub("/#{Hyrax::GenericWorksController::IIIF_DEFAULT_VERSION}/", "/2/") }
+    let(:mime3) { Hyrax::GenericWorksController::IIIF_DEFAULT_MANIFEST_MIME.gsub("/#{Hyrax::GenericWorksController::IIIF_DEFAULT_VERSION}/", "/3/") }
+    let(:manifest_factory2) { instance_double("IIIFManifest::ManifestBuilder", to_h: { test: 'manifest2' }) }
+    let(:manifest_factory3) { instance_double("IIIFManifest::V3::ManifestBuilder", to_h: { test: 'manifest3' }) }
+
+    before do
+      allow(controller).to receive(:presenter).and_return(presenter)
+      allow(IIIFManifest::ManifestFactory).to receive(:new)
+        .with(Hyrax::WorkShowPresenter)
+        .and_return(manifest_factory2)
+      allow(IIIFManifest::V3::ManifestFactory).to receive(:new)
+        .with(Hyrax::WorkShowPresenter)
+        .and_return(manifest_factory3)
+      request.headers['Accept'] = mime
+    end
+
+    context 'for request accepting IIIF V2' do
+      let(:mime) { mime2 }
+
+      it 'returns IIIF V2 manifest' do
+        get :manifest, params: { id: 'testwork', format: :json }
+        expect(response.headers['Content-Type']).to eq mime
+        expect(response.body).to eq "{\"test\":\"manifest2\"}"
+      end
+      # end
+    end
+
+    context 'for request accepting IIIF V3' do
+      let(:mime) { mime3 }
+
+      it 'returns IIIF V3 manifest' do
+        get :manifest, params: { id: 'testwork', format: :json }
+        expect(response.headers['Content-Type']).to eq mime
+        expect(response.body).to eq "{\"test\":\"manifest3\"}"
+      end
+    end
+
+    context 'for request without IIIF profile' do
+      let(:mime) { "application/json;" }
+
+      it 'returns IIIF default version manifest' do
+        get :manifest, params: { id: 'testwork', format: :json }
+        expect(response.headers['Content-Type']).to eq Hyrax::GenericWorksController::IIIF_DEFAULT_MANIFEST_MIME
+        if Hyrax::GenericWorksController::IIIF_DEFAULT_VERSION == 2
```

Comment:
This conditional makes the test resilient to the change of the `IIIF_DEFAULT_VERSION` from 2 to 3, but since the controller code is set to 2 I'd expect this test to only cover the Presentation 2 case and when that constant changes then this test would change too.